py-utils/src/utils/setup/consul/consul.py

```diff
@@ -0,0 +1,227 @@
+
+        command = "systemctl daemon-reload"
+        _, err, returncode = SimpleProcess(command).run()
+        if returncode != 0:
+            ConsulSetupError(returncode, "Consul Setup systemd daemon-reload failed with error: %s" % err)
+        return 0
+
+    def init(self):
+        """ Perform initialization. Raises exception on error """
+
+        # TODO: Perform actual steps. Obtain inputs using Conf.get(index, ..)
+        return 0
+
+    def config(self):
+        """ Performs configurations. Raises exception on error """
+
+        bind_addr = Conf.get(self.index, f"server_node>{Conf.machine_id}>network>data>private_interfaces[0]")
+        config_path = Conf.get(self.index, "cortx>software>consul>config_path", "/etc/consul.d")
+        data_path = Conf.get(self.index, "cortx>software>consul>data_path", "/opt/consul")
+        self.cluster_id = Conf.get(self.index, f"server_node>{Conf.machine_id}>cluster_id")
+        server_node_fqdns = []
+        storage_sets = Conf.get(self.index, f"cluster>{self.cluster_id}>storage_set")
+        for storage_set in storage_sets: 
+            for server_node in storage_set["server_nodes"]:
+                if self.run_as_server(server_node) and server_node != Conf.machine_id:
+                    server_node_fqdns.append(
+                        Conf.get(self.index, f"server_node>{server_node}>network>data>private_fqdn")
+                    )
+
+        server = self.run_as_server(Conf.machine_id)
+        env = Environment(loader=FileSystemLoader("/opt/seagate/cortx/utils/conf"))
+        template = env.get_template('consul.hcl.tmpl')
+        template.stream(bind_addr=bind_addr,
+                        data_dir=data_path,
+                        retry_join=json.dumps(server_node_fqdns),
+                        server=str(server).lower(),
+                        bootstrap_expect=2).dump(f"{config_path}/consul.hcl")
+        command = f"consul validate {config_path}/consul.hcl"
+        _, err, returncode = SimpleProcess(command).run()
+        if returncode != 0:
+            ConsulSetupError(returncode, 
+                            "Consul Setup config file %s validation failed with error :%s"
+                            %(f"{config_path}/consul.hcl", err))
+        command = f"chown -R consul:consul {config_path} {data_path}"
+        _, err, returncode = SimpleProcess(command).run()
+        if returncode != 0:
+            ConsulSetupError(returncode, 
+                            "Consul Setup changing ownership failed for %s %s with error: %s"
+                            %(config_path, data_path, err))
+        return 0
+
+    def run_as_server(self, server_node):
+        storage_set_count = int(Conf.get(self.index, f"cluster>{self.cluster_id}>site>storage_set_count"))
+
+        node_id = int(Conf.get(self.index, f"server_node>{server_node}>node_id"))
+        if storage_set_count < 5:
+            if node_id < 4:
+                return True
```

Comment:
Let's keep it simple.
```suggestion
            return node_id < 4
```